pkg/assetvalidator/errors.go

```diff
@@ -2,18 +2,36 @@ package assetvalidator
 
 import "strings"
 
-// MultiValidatorError occurs when any validation check has an error.
-// MultiValidatorError keeps track of a list of errors.
-type MultiValidatorError struct {
+// ValidationError occurs when the process of validation fails
+// unexpectantly, and not due to an invalid attributed asset
+// Examples of usage would include a bad http call
+type ValidationError struct {
```

Comment:
Task: update the `ValidationFailure` and `ValidationError` error types to align with what we are doing with `domain.AssetNotFoundError`.

Details:

Now that we are making the decision to only log `ValidationFailure`, meaning the validation operation completed with a failure result, let's add the structure we need to handle them properly in the HTTP handler.

After doing some digging, I think we should follow the error handling pattern we used with `domain.AssetNotFoundError` in this service, which is:

`ValidationFailure` has an `Inner` field  of type `error` _interface_ that contains any wrapped error that will help build out its eventual error message. The key part is that this wrapped error can be anything, including another individual `ValidationFailure` error, or a new error implementation that has an error slice to collect multiples, along with its own `Error` receiver function to use them to construct their part of the error message.

In this implementation, the `ValidationFailure` error would also contain fields for `Inner` for the wrapped error, `AssetID` to identify the event payload, and `FailedCheck` that would contain a log friendly name for the check that failed. This type would then have its `Error` func that uses those fields to compose a readable error message.

Then, you'd have a likely unexported error type, `multiValidationFailure` that has a slice for collecting more than one validation failure, and generating the rest of the error message from the combination of all of them. Something kind of like this:

```
// multiValidationFailure combines multiple error strings in a single message
type multiValidationFailure struct {
	Failures []error
}

func (f multiValidationFailure) Error() string {
	return fmt.Sprintf("failures: %v", f.Failures)
}
```

Does that make sense?

Reply:
Does it make sense for `ValidationFailure` to have AssetID? wouldn't that be redundant for embedded `ValidationFailure`?

Reply:
This is kinda confusing... I don't see why this pattern is better than what I have on this PR? Doesn't this achieve the same thing, and it's shorter?

Reply:
I commented below about where and why this implementation can actually become more confusing over time, due to leaking details between implementations . I can see your point about the Asset ID, but I think the benefits outweigh the costs.